src/newsimplification/Simplifier.ts

```diff
@@ -0,0 +1,150 @@
+import { CommonModel, Schema } from '../models';
+import { SimplificationOptions } from '../models/SimplificationOptions';
+import { simplifyName, isModelObject } from './Utils';
+import simplifyProperties from './SimplifyProperties';
+import simplifyAllOf from './SimplifyAllOf';
+
+export class Simplifier {
+  static defaultOptions: SimplificationOptions = {
+    allowInheritance: false
+  }
+
+  private anonymCounter = 1;
+  private seenSchemas: Map<Schema, CommonModel> = new Map();
+  private iteratedModels: Record<string, CommonModel> = {};
+  
+  constructor(
+    readonly options: SimplificationOptions = Simplifier.defaultOptions,
+  ) {
+    this.options = { ...Simplifier.defaultOptions, ...options };
```

Comment:
It is ok for now. But if you would add nested options you would need to change it anyway.

Reply:
You are right, but we will tackle it then if that happens, thanks for the headsup!

Reply:
In future we can use deep-merge or write simple function for it, but as you wrote, currently it's enough. If someone add some nested option, then should also update this line.